src/ws-api.ts

```diff
@@ -1,29 +1,24 @@
-import { DataStream, type Dwn } from '@tbd54566975/dwn-sdk-js';
 
-import type { IncomingMessage, Server } from 'http';
-import { base64url } from 'multiformats/bases/base64';
-import { v4 as uuidv4 } from 'uuid';
-import { type AddressInfo, type WebSocket, WebSocketServer } from 'ws';
+import type {
+  Dwn,
+} from '@tbd54566975/dwn-sdk-js';
 
-import type { RequestContext } from './lib/json-rpc-router.js';
-import {
-  createJsonRpcErrorResponse,
-  JsonRpcErrorCodes,
-  type JsonRpcResponse,
-} from './lib/json-rpc.js';
+import type { AddressInfo } from 'ws';
+import type { Server } from 'http';
 
-import { jsonRpcApi } from './json-rpc-api.js';
-import { requestCounter } from './metrics.js';
+import { WebSocketServer } from 'ws';
 
-const SOCKET_ISALIVE_SYMBOL = Symbol('isAlive');
-const HEARTBEAT_INTERVAL = 30_000;
+import type { ConnectionManager } from './connection/connection-manager.js';
+import { InMemoryConnectionManager } from './connection/connection-manager.js';
 
 export class WsApi {
   #wsServer: WebSocketServer;
   dwn: Dwn;
+  #connections: ConnectionManager
 
-  constructor(server: Server, dwn: Dwn) {
+  constructor(server: Server, dwn: Dwn, connectionManager?: ConnectionManager) {
     this.dwn = dwn;
+    this.#connections = connectionManager || new InMemoryConnectionManager(dwn);
     this.#wsServer = new WebSocketServer({ server });
   }
 
```

Comment:
Nit: line 25 `get address()` appears to be of not much use since `server` itself is exposed, consider removing as it is only serving a test at the moment...

Reply:
Done